Mathlib/LinearAlgebra/QuadraticForm/Basis.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2024 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.LinearAlgebra.QuadraticForm.Basic
+import Mathlib.Data.Finset.Sym
+import Mathlib.Data.Sym.Sym2.Order
+
+/-!
+# Constructing a bilinear form from a quadratic form, given a basis
+
+This file provides an alternative to `QuadraticForm.associated`; unlike that definition, this one
+does not require `Invertible (2 : R)`. Unlike that definition, this only works in the presence of
+a basis.
+-/
+
+-- TODO: move
+theorem Finset.sum_sym2_filter_not_isDiag {ι α} [LinearOrder ι] [AddCommMonoid α]
+    (s : Finset ι) (p : Sym2 ι → α) :
+    ∑ i in s.sym2.filter (¬ ·.IsDiag), p i =
+      ∑ i in s.offDiag.filter (fun i => i.1 < i.2), p s(i.1, i.2) := by
+  rw [Finset.offDiag_filter_lt_eq_filter_le]
+  conv_rhs => rw [← Finset.sum_subtype_eq_sum_filter]
+  refine (Finset.sum_equiv Sym2.sortEquiv.symm ?_ ?_).symm
+  · rintro ⟨⟨i₁, j₁⟩, hij₁⟩
+    simp [and_assoc]
+  · rintro ⟨⟨i₁, j₁⟩, hij₁⟩
+    simp
```

Comment:
`Algebra.Order.BigOperators.Group.Finset` should do

Reply:
Let's see if this blows up the number of imports...